common/utils.js

```diff
@@ -188,11 +297,12 @@ export function getInputTensor(inputElement, inputOptions) {
           value = pixels[h * width * imageChannels + w * imageChannels + c];
         }
         if (inputLayout === 'nchw') {
-          tensor[c * width * height + h * width + w] =
-              (value - mean[c]) / std[c];
+          tensor[c * width * height + h * width + w] = dataType === 'float16' ?
+              toHalf((value - mean[c]) / std[c]) : (value - mean[c]) / std[c];
```

Comment:
Converting float32 to float16 in JS is slow, try WebNN `cast` operator.

Reply:
Done, thanks!